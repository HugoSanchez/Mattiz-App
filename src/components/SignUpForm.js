import React, { Component } from 'react';
import { View, ImageBackground, StyleSheet } from 'react-native';
import {  Button, Input } from 'react-native-elements';
import LinearGradient from 'react-native-linear-gradient';

import { connect } from 'react-redux';

import { authCreateUser, setTokenInMemory } from '../api/auth';
import { setUser } from '../actions/AuthActions';

class SignUpForm extends Component {

    state = {
        username: null,
        password: null,
        confirmPassword: null,
        formStatus: false,
        isFocused: false,
        loading: false
    }

    onButtonPress() {
        // Deconstruct state.
        const { username, password, confirmPassword } = this.state;
        // Check passwords are correct.
        if ( password == confirmPassword ) {
            // If so, call the '/register' endpoint which returns token.
            authCreateUser(username, password).then(res => {
                if (res.data.auth) {
                    // If token, save it in memory
                    setTokenInMemory(res.data.token)
                    // And set user in redux state.
                    setUser({username})
                    // Navigate user inside app. 
                    this.props.navigation.navigate('Dashboard')
                }
            })
        }
    }

    render() {
        const {container, input, signUpButton, viewStyle } = styles;
        if ( this.state.formStatus == false ) {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/LoginBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <Button 
                        title='Sign Up !'
                        titleStyle={{ color: '#03001A' }}
                        buttonStyle={ signUpButton }
                        onPress={() => this.setState({ formStatus: 'username' }) }
                        ViewComponent={LinearGradient}
                        linearGradientProps={{
                            colors: ['rgba(214, 213, 213, 1)', 'rgba(163, 209, 100, .4)'],
                            start: { x: 0.3, y: 0.5 },
                            end: { x:1, y: 0.5 },
                        }}
                    />
                    </ImageBackground>
                </View>
            )
        } 
        
        //*  SET USERNAME   *//
        if ( this.state.formStatus === 'username') {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/LoginBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={ viewStyle }>
                        <Input 
                            placeholder="What's your name?"
                            inputContainerStyle={{ borderBottomColor: 'transparent', marginTop: 5 }}
                            placeholderTextColor='gray'
                            value={this.state.username}
                            onChangeText={ value => this.setState({ username: value }) }
                            fontStyle={ this.state.password? 'italic' : 'normal' }
                            onFocus={() => this.setState({ isFocused: true })}
                            style={ input }
                        />
                    </View>
                    <Button 
                        title='Next !'
                        titleStyle={{ color: '#03001A' }}
                        buttonStyle={{ margin: 12, marginTop: this.state.isFocused ? 10 : 347, height: 47, backgroundColor: 'transparent' }}
                        onPress={ () => this.setState({ formStatus: 'password' }) }
                        ViewComponent={LinearGradient}
                        linearGradientProps={{
                            colors: ['rgba(214, 213, 213, 1)', 'rgba(163, 209, 100, .4)'],
                            start: { x: 0.3, y: 0.5 },
                            end: { x:1, y: 0.5 },
                        }}
                    />
                    </ImageBackground>
                </View>
            )
        }

        //*  SET PASSWORD   *//
        if ( this.state.formStatus === 'password') {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/LoginBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={ viewStyle }>
                        <Input 
                            placeholder="Choose your Password"
                            inputContainerStyle={{ borderBottomColor: 'transparent', marginTop: 5 }}
                            placeholderTextColor='gray'
                            value={this.state.password}
                            secureTextEntry={true}
                            onChangeText={ value => this.setState({ password: value }) }
                            fontStyle={ this.state.password ? 'italic' : 'normal' }
                            onFocus={() => this.setState({ isFocused: true })}
                            style={ input }
                        />
                    </View>
                    <Button 
                        title='Next !'
                        titleStyle={{ color: '#03001A' }}
                        buttonStyle={{ margin: 12, marginTop: this.state.isFocused ? 10 : 347, height: 47, backgroundColor: 'transparent' }}
                        onPress={ () => this.setState({ formStatus: 'confirm' }) }
                        ViewComponent={LinearGradient}
                        linearGradientProps={{
                            colors: ['rgba(214, 213, 213, 1)', 'rgba(163, 209, 100, .4)'],
                            start: { x: 0.3, y: 0.5 },
                            end: { x:1, y: 0.5 },
                        }}
                    />
                    </ImageBackground>
                </View>
            )
        }

        //*  CONFIRM PASSWORD   *//
        if ( this.state.formStatus === 'confirm') {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/LoginBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={ viewStyle }>
                        <Input 
                            placeholder="Confirm your Password"
                            inputContainerStyle={{ borderBottomColor: 'transparent', marginTop: 5 }}
                            placeholderTextColor='gray'
                            value={this.state.confirmPassword}
                            secureTextEntry={true}
                            onChangeText={ value => this.setState({ confirmPassword: value }) }
                            fontStyle={ this.state.confirmPassword ? 'italic' : 'normal' }
                            onFocus={() => this.setState({ isFocused: true })}
                            style={ input }
                        />
                    </View>
                    <Button 
                        title='Sign Up !'
                        titleStyle={{ color: '#03001A' }}
                        buttonStyle={{ margin: 12, marginTop: this.state.isFocused ? 10 : 347, height: 47, backgroundColor: 'transparent' }}
                        onPress={ this.onButtonPress.bind(this) }
                        ViewComponent={LinearGradient}
                        linearGradientProps={{
                            colors: ['rgba(214, 213, 213, 1)', 'rgba(163, 209, 100, .4)'],
                            start: { x: 0.3, y: 0.5 },
                            end: { x:1, y: 0.5 },
                        }}
                    />
                    </ImageBackground>
                </View>
            )
        }
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1, 
        justifyContent: 'center',
        backgroundColor: '#03001A'
    },
    input: {
        height: 50,
        backgroundColor: '#ededed',
        margin: 30,
        padding: 20
    },
    viewStyle: {
        marginTop: 281, 
        backgroundColor: 'rgba(255, 255, 255, 0.8)', 
        borderRadius: 2,
        margin: 12, 
        height: 45
    },
    signUpButton: {
        marginTop: 700,
        margin: 12, 
        backgroundColor: 'transparent',
        height: 45
    }
})

const MapStateToProps = state => {
    const { user, token, error } = state.auth;
    return {
        user,
        token,
        error
    };
}

export default connect(MapStateToProps, { setUser })(SignUpForm);
