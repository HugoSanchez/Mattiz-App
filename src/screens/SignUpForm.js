import React, { Component } from 'react';
import { View, ImageBackground, StyleSheet } from 'react-native';
import { Input } from 'react-native-elements';

// Custom Components.
import MattizButton from '../components/common/MattizButton';

// Connect.
import { connect } from 'react-redux';

// Redux Actions and Auth Functions. 
import { authCreateUser, setTokenInMemory } from '../api/auth';
import { setUserInReduxState } from '../actions';

class SignUpForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: null,
            password: null,
            confirmPassword: null,
            formStatus: false,
            isFocused: false,
            loading: false
        }
    }

    onButtonPress() {
        // Deconstruct state.
        const { username, password, confirmPassword } = this.state;
        // Check if passwords are correct.
        if ( password == confirmPassword ) {
            // If so, call the '/register' endpoint which returns token.
            authCreateUser(username, password).then(res => {
                if (res.data.auth) {
                    // If token, save it in memory
                    setTokenInMemory('token', res.data.token)
                    // And set user in redux state.
                    this.props.setUserInReduxState(res.data.user)
                    // Navigate user to onboarding set up. 
                    this.props.navigation.navigate('Welcome')
                }
            })
        }
    }

    render() {
        const { container, input, viewStyle } = styles;

        if ( this.state.formStatus == false ) {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/whiteBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={{ marginTop: 700 }}>
                        <MattizButton 
                            title={'Set Sail !'}
                            titleStyle={{ color: '#040026'}}
                            onPress={ () => this.setState({ formStatus: 'username' })}
                            linearColor={'rgba(214, 213, 213, 1)'}
                        />
                    </View>
                    </ImageBackground>
                </View>
            );
        } 
        
        //  SET USERNAME   //
        if ( this.state.formStatus === 'username') {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/whiteBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={ viewStyle }>
                        <Input 
                            placeholder="What's your name?"
                            inputContainerStyle={{ borderBottomColor: 'transparent', marginTop: 5 }}
                            placeholderTextColor='gray'
                            value={this.state.username}
                            onChangeText={ value => this.setState({ username: value }) }
                            fontStyle={ this.state.password? 'italic' : 'normal' }
                            onFocus={() => this.setState({ isFocused: true })}
                            style={ input }
                        />
                    </View>
                    <View style={{ marginTop: this.state.isFocused ? 10 : 347}}>
                        <MattizButton 
                            title={'Next !'}
                            titleStyle={{ color: '#040026'}}
                            onPress={ () => this.setState({ formStatus: 'password' })}
                            linearColor={'rgba(214, 213, 213, 1)'}
                        />
                    </View>
                    </ImageBackground>
                </View>
            );
        }

        //  SET PASSWORD   //
        if ( this.state.formStatus === 'password') {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/whiteBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={ viewStyle }>
                        <Input 
                            placeholder="Choose your Password"
                            inputContainerStyle={{ borderBottomColor: 'transparent', marginTop: 5 }}
                            placeholderTextColor='gray'
                            value={this.state.password}
                            secureTextEntry={true}
                            onChangeText={ value => this.setState({ password: value }) }
                            fontStyle={ this.state.password ? 'italic' : 'normal' }
                            onFocus={() => this.setState({ isFocused: true })}
                            style={ input }
                        />
                    </View>
                    <View style={{ marginTop: this.state.isFocused ? 10 : 347}}>
                        <MattizButton 
                            title={'Next !'}
                            titleStyle={{ color: '#040026'}}
                            onPress={ () => this.setState({ formStatus: 'confirm' })}
                            linearColor={'rgba(214, 213, 213, 1)'}
                        />
                    </View>
                    </ImageBackground>
                </View>
            );
        }

        //   CONFIRM PASSWORD   //
        if ( this.state.formStatus === 'confirm') {
            return (
                <View style={ container }>
                    <ImageBackground 
                        source={require('../assets/whiteBackground.png')} 
                        style={{ width:'100%', height:'100%', flex: 1 }}
                        resizeMode='cover'
                    >
                    <View style={ viewStyle }>
                        <Input 
                            placeholder="Confirm your Password"
                            inputContainerStyle={{ borderBottomColor: 'transparent', marginTop: 5 }}
                            placeholderTextColor='gray'
                            value={this.state.confirmPassword}
                            secureTextEntry={true}
                            onChangeText={ value => this.setState({ confirmPassword: value }) }
                            fontStyle={ this.state.confirmPassword ? 'italic' : 'normal' }
                            onFocus={() => this.setState({ isFocused: true })}
                            style={ input }
                        />
                    </View>
                    <View style={{ marginTop: this.state.isFocused ? 10 : 347}}>
                        <MattizButton 
                            title={'Sign Up !'}
                            titleStyle={{ color: '#040026'}}
                            onPress={ this.onButtonPress.bind(this) }
                            linearColor={'rgba(214, 213, 213, 1)'}
                        />
                    </View>
                    </ImageBackground>
                </View>
            );
        }
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1, 
        justifyContent: 'center',
        backgroundColor: '#03001A'
    },
    input: {
        height: 50,
        backgroundColor: '#ededed',
        margin: 30,
        padding: 20
    },
    viewStyle: {
        marginTop: 281, 
        backgroundColor: 'rgba(255, 255, 255, 0.8)', 
        borderRadius: 2,
        margin: 12, 
        height: 45
    }
})

const MapStateToProps = state => {
    const { user, token, error } = state.auth;
    return {
        user,
        token,
        error
    };
}

export default connect(MapStateToProps, { setUserInReduxState })(SignUpForm);
