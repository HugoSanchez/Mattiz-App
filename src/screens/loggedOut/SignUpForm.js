import React, {Component} from 'react'
import {
	View,
	Dimensions,
	ImageBackground,
	Image,
	StyleSheet,
} from 'react-native'
import {Input} from 'react-native-elements'

// Custom Components.
import PalmButton from '../../components/common/PalmButton'
import LoadingScreen from '../../components/LoadingScreen'

// Connect.
import {connect} from 'react-redux'

// Redux Actions and Auth Functions.
import {authCreateUser, setTokenInMemory} from '../../api/auth'
import {setUserInReduxState} from '../../actions'
import colors from '../../constants/colors'

// Dims.
const width = Dimensions.get('window').width
const height = Dimensions.get('window').height

class SignUpForm extends Component {
	constructor(props) {
		super(props)
		this.state = {
			username: null,
			password: null,
			confirmPassword: null,
			formStatus: false,
			isFocused: false,
			loading: false,
		}
	}

	async componentWillMount() {
		/**
        const sdk = new UniversalLoginSDK('http://192.168.1.35:3311', "http://192.168.1.35:18545");
        console.log('Hello')

        const [privateKey, contractAddress] = await sdk.create('pepito.mylogin.eth');
        console.log('Private Key: ', privateKey)
        console.log('Contract Address: ', contractAddress) 
         */
	}

	onButtonPress() {
		// Deconstruct state.
		const {username, password, confirmPassword} = this.state
		// Check if passwords are correct.
		if (password == confirmPassword) {
			// Set loading == true.
			this.setState({loading: true})
			// If so, call the '/register' endpoint which returns token.
			authCreateUser(username, password).then(res => {
				if (res.data.auth) {
					console.log('Res: ', res.data)
					// If token, save it in memory
					setTokenInMemory('token', res.data.token)
					// And set user in redux state.
					this.props.setUserInReduxState(res.data.user)
					// Navigate user to onboarding set up.
					this.props.navigation.navigate('Welcome')
				}
			})
		}
	}

	render() {
		const {
			container,
			logoContainer,
			input,
			viewStyle,
			textInput,
			placeholder,
			imageStyle,
		} = styles

		if (this.state.loading) {
			return (
				<View style={container}>
					<LoadingScreen />
				</View>
			)
		}

		if (this.state.formStatus == false) {
			return (
				<View style={container}>
					<ImageBackground
						source={require('../../assets/Background.png')}
						style={{width: '100%', height: '100%', flex: 1}}
						resizeMode={'cover'}>
						<View style={logoContainer}>
							<Image
								source={require('../../assets/Mattiz-Didot-Simple-Squared-Green.png')}
								style={imageStyle}
							/>
						</View>

						<View
							style={{
								position: 'absolute',
								bottom: '2%',
								width: '95%',
								alignSelf: 'center',
							}}>
							<PalmButton
								backgroundColor={colors.palleteLightGreen}
								onPress={() =>
									this.setState({formStatus: 'username'})
								}
								title={'Sign Up'}
							/>
						</View>
					</ImageBackground>
				</View>
			)
		}

		//  SET USERNAME   //
		if (this.state.formStatus === 'username') {
			return (
				<View style={container}>
					<ImageBackground
						source={require('../../assets/Background.png')}
						style={{width: '100%', height: '100%', flex: 1}}
						resizeMode={'cover'}>
						<View style={logoContainer}>
							<Image
								source={require('../../assets/Mattiz-Didot-Simple-Squared-Green.png')}
								style={imageStyle}
							/>
						</View>
						<View style={viewStyle}>
							<Input
								testID="SignUpFormNameInput"
								placeholder="What's your name?"
								placeholderTextColor={colors.darkGray}
								value={this.state.username}
								onChangeText={value =>
									this.setState({username: value})
								}
								fontStyle={'italic'}
								onFocus={() => this.setState({isFocused: true})}
								inputStyle={
									this.state.username
										? textInput
										: placeholder
								}
								style={input}
							/>
						</View>
						<View
							style={{
								alignSelf: 'center',
								position: 'absolute',
								top: this.state.isFocused ? '45%' : '90%',
							}}>
							<PalmButton
								iconName={'arrow-right'}
								onPress={() =>
									this.setState({formStatus: 'password'})
								}
								title={'Continue'}
							/>
						</View>
					</ImageBackground>
				</View>
			)
		}

		//  SET PASSWORD   //
		if (this.state.formStatus === 'password') {
			return (
				<View style={container}>
					<ImageBackground
						source={require('../../assets/Background.png')}
						style={{width: '100%', height: '100%', flex: 1}}
						resizeMode={'cover'}>
						<View style={logoContainer}>
							<Image
								source={require('../../assets/Mattiz-Didot-Simple-Squared-Green.png')}
								style={imageStyle}
							/>
						</View>
						<View style={viewStyle}>
							<Input
								placeholder="Choose your password"
								placeholderTextColor="#464646"
								value={this.state.password}
								secureTextEntry={true}
								onChangeText={value =>
									this.setState({password: value})
								}
								fontStyle={'italic'}
								inputStyle={
									this.state.password
										? textInput
										: placeholder
								}
								onFocus={() => this.setState({isFocused: true})}
								style={input}
							/>
						</View>
						<View
							style={{
								alignSelf: 'center',
								position: 'absolute',
								top: this.state.isFocused ? '45%' : '90%',
							}}>
							<PalmButton
								iconName={'arrow-right'}
								onPress={() =>
									this.setState({formStatus: 'confirm'})
								}
								title={'Continue'}
							/>
						</View>
					</ImageBackground>
				</View>
			)
		}

		//   CONFIRM PASSWORD   //
		if (this.state.formStatus === 'confirm') {
			return (
				<View style={container}>
					<ImageBackground
						source={require('../../assets/Background.png')}
						style={{width: '100%', height: '100%', flex: 1}}
						resizeMode={'cover'}>
						<View style={logoContainer}>
							<Image
								source={require('../../assets/Mattiz-Didot-Simple-Squared-Green.png')}
								style={imageStyle}
							/>
						</View>
						<View style={viewStyle}>
							<Input
								placeholder="Confirm password"
								placeholderTextColor="#464646"
								value={this.state.confirmPassword}
								secureTextEntry={true}
								onChangeText={value =>
									this.setState({confirmPassword: value})
								}
								fontStyle={'italic'}
								inputStyle={
									this.state.confirm ? textInput : placeholder
								}
								onFocus={() => this.setState({isFocused: true})}
								style={input}
							/>
						</View>
						<View
							style={{
								alignSelf: 'center',
								position: 'absolute',
								top: this.state.isFocused ? '45%' : '90%',
							}}>
							<PalmButton
								iconName={'check-circle'}
								onPress={() => this.onButtonPress()}
								title={'Sign up!'}
							/>
						</View>
					</ImageBackground>
				</View>
			)
		}
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		backgroundColor: '#FFF',
	},
	logoContainer: {
		position: 'absolute',
		top: '10%',
		width: '95%',
		alignSelf: 'center',
	},
	viewStyle: {
		position: 'absolute',
		top: '40%',
		flexDirection: 'row',
		marginLeft: '10%',
		marginRight: '10%',
		height: '5%',
		borderBottomWidth: 1,
		borderBottomColor: 'gray',
	},
	placeholder: {
		fontFamily: 'Rajdhani-Light',
		fontSize: 18,
	},
	textInput: {
		fontFamily: 'Rajdhani-Regular',
		fontSize: 20,
		color: colors.primaryBlue,
	},
	imageStyle: {
		width: 180,
		height: 140,
		alignSelf: 'center',
	},
})

const mapDispatchtoProps = dispatch => ({
	setUserInReduxState: user => dispatch(setUserInReduxState(user)),
})

const mapStateToProps = state => {
	const {user, token, error} = state.auth
	return {
		user,
		token,
		error,
	}
}

export default connect(
	mapStateToProps,
	mapDispatchtoProps,
)(SignUpForm)
