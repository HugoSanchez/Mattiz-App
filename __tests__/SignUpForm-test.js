import 'react-native'
import configureMockStore from 'redux-mock-store'
import React from 'react'
import SignUpForm from '../src/screens/loggedOut/SignUpForm'
import {Provider} from 'react-redux'
import {render, fireEvent} from '@testing-library/react-native'

const mockStore = configureMockStore([])
store = mockStore({
	auth: {user: {}, token: 'token', error: false},
})

test('SignUpForm component', () => {
	const {
		queryByText,
		queryByPlaceholderText,
		queryByTestId,
		container,
	} = render(
		<Provider store={store}>
			<SignUpForm />
		</Provider>,
	)

	fireEvent.press(queryByText('Sign Up'))
	const userNameInput = queryByPlaceholderText("What's your name?")

	expect(userNameInput).toBeTruthy()
	expect(queryByText('Sign Up')).toBeNull()

	fireEvent.changeText(userNameInput, 'Marju')
	fireEvent.press(queryByText('Continue'))
	const passwordInput = queryByPlaceholderText('Choose your password')

	expect(passwordInput).toBeTruthy()

	fireEvent.changeText(passwordInput, '42')
	fireEvent.press(queryByText('Continue'))
	const confirmInput = queryByPlaceholderText('Confirm password')

	expect(confirmInput).toBeTruthy()

	fireEvent.changeText(confirmInput, '42')
	fireEvent.press(queryByText('Sign up!'))

	expect(queryByText('Sign up!')).toBeNull()
	expect(queryByTestId('LoadingScreen')).toBeTruthy()
	expect(container.children[0]).toMatchSnapshot()
})
