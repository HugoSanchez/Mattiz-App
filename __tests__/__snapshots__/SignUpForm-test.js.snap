// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="LoadingScreen"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RNSpinkit
      color={4279582803}
      size={75}
      style={
        Array [
          Object {
            "height": 75,
            "width": 75,
          },
          Object {
            "alignSelf": "center",
          },
        ]
      }
      type="Pulse"
    />
  </View>
</View>
`;
